backend/src/services/comparisons/comparisons.service.ts

```diff
@@ -0,0 +1,194 @@
+import { prisma } from '@data/prisma-client';
+import { Comparison, ComparisonType, Complectation } from '@prisma/client';
+
+const addCarToComparison = async ({
+  complectationId,
+  userId,
+}: {
+  complectationId: string;
+  userId: string;
+}): Promise<Comparison> => {
+  const activeComparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+  });
+
+  if (activeComparison) {
+    const presenceCheck = await prisma.comparisons_Complectations.findFirst({
+      where: { complectation_id: complectationId },
+    });
+
+    if (presenceCheck) {
+      throw new Error('Car is already in your active comparison');
+    }
+
+    const comparison = await prisma.comparison.update({
+      where: { id: activeComparison.id },
+      data: { updated_at: new Date() },
+    });
+
+    const carsInComparisonAmount =
+      await prisma.comparisons_Complectations.count({
+        where: { comparison_id: comparison.id },
+      });
+
+    await prisma.comparisons_Complectations.create({
+      data: {
+        comparison_id: comparison.id,
+        complectation_id: complectationId,
+        position: carsInComparisonAmount + 1,
+      },
+    });
+
+    return comparison;
+  }
+
+  if (!activeComparison) {
```

Comment:
Why do we need this check? :slightly_smiling_face: (prev condition allows us to safely suggest that the `activeComparison` doesn't exist)